### ISSUE TITLE
should mask specified secret values within log messages

### ISSUE THREAD

**Issue by hoatle** [teracyhq/dev#492]
Some values are sensitive and should be masked on log messages, we should allow masking these values before displaying it on the console.

By doing this, we can ask users to provide debug logs safely for easier support and debug.

~~I guess we can have some config like this:~~

```yaml
logger:
  masked_values: []
```

updated: follow my comments from here https://github.com/teracyhq/dev/pull/551#issuecomment-438885667 to see in detail.

Basically, if teracy-dev-entry/logger.yaml is available with the follow format:

```yaml
logger:
  mask:
    variables: {}
    values: []
```
the mask should work.

**Comment by hoatle** [comment]
LogMasker should provide APIs for manipulating the masked_values

**Comment by hoatle** [comment]
assigned to @hieptranquoc

**Comment by hoatle** [comment]
@hieptranquoc there is another bug like this:
- create logger.yaml with empty content
- create config_override.yaml with variables and its value
- create log that having the values from variables of config_override.yaml

expected: no mask
actual: all the variables values were masked

note: we should only resolve the variables defined within the logger

logger.yaml
```yaml
logger:
  mask:
    variables:
      entry_secret: ${ENTRY_SECRET:-abcdef}
```

config_override.yaml
```
variables:
  entry_secret: ${ENTRY_SECRET:-hi}
  another_var: hello
```

logger.debug("test hello abcdef hi")

then `$ LOG_LEVEL=debug vagrant status`
- expected: "test hello abcdef **"
- actual: "test ***** abcdef **"

**Comment by hoatle** [comment]
bug when there is no logger.yaml file:

```
$ vagrant status
Vagrant failed to initialize at a very early stage:

There was an error loading a Vagrantfile. The file being loaded
and the error message are shown below. This is usually caused by
a syntax error.

Path: /Users/hoatle/teracy-dev/Vagrantfile
Line number: 54
Message: NoMethodError: undefined method `[]' for nil:NilClass
```

**Comment by hoatle** [comment]
fixed by: https://github.com/teracyhq/dev/commit/8ab407b076fedd3208b9fa2b47e5b20f3ffce845
